pkg/service/auth_context.go

```diff
@@ -3,54 +3,119 @@ package service
 import (
 	"fmt"
 	"strings"
+	"sync"
 
 	"github.com/3scale/authorino/pkg/config"
+	"github.com/3scale/authorino/pkg/config/internal"
 
-	"golang.org/x/net/context"
 	auth "github.com/envoyproxy/go-control-plane/envoy/service/auth/v2"
+	"golang.org/x/net/context"
 )
 
 // AuthContext holds the context of each auth request, including the request itself (sent by the client),
 // the auth config of the requested API and the lists of identity verifications, metadata add-ons and
 // authorization policies, and their corresponding results after evaluated
 type AuthContext struct {
 	ParentContext *context.Context
-	Request *auth.CheckRequest
-	API *APIConfig
+	Request       *auth.CheckRequest
+	API           *APIConfig
+
+	Identity      map[*config.IdentityConfig]interface{}
+	Metadata      map[*config.MetadataConfig]interface{}
+	Authorization map[*config.AuthorizationConfig]interface{}
+}
 
-	Identity map[*config.IdentityConfig] interface{}
-	Metadata map[*config.MetadataConfig] interface{}
-	Authorization map[*config.AuthorizationConfig] interface{}
+// AuthObjectConfig provides an interface for APIConfig objects that implements a Call method
+type AuthObjectConfig interface {
+	Call(ctx internal.AuthContext) (interface{}, error)
+}
+
+func (authContext *AuthContext) getAuthObject(authObjConfig AuthObjectConfig) (interface{}, error) {
+	if ret, err := authObjConfig.Call(authContext); err != nil {
```

Comment:
I think this method does not add any value. In any case, you could just return the value of the called method
```
func (authContext *AuthContext) getAuthObject(authObjConfig AuthObjectConfig) (interface{}, error) {  
    return authObjConfig.Call(authContext)                                                                                                 
}                                                                                                     
```

Reply:
yeah, this one got slim after last refactor... it doesn't make sense now, thanks!